Cloudinary/Classes/Core/CLDConfiguration.swift

```diff
@@ -297,71 +311,87 @@ import Foundation
             secureDistribution = String(uri.path[index1...])
         }
         
-        if let params = uri.query?.components(separatedBy: "&") {
-            for param in params {
-                let keyValue = param.components(separatedBy: "=")
-                if keyValue.count < 2 {
+        if let queryItems = URLComponents(url: uri, resolvingAgainstBaseURL: false)?.queryItems {
+            for item in queryItems
+            {
+                guard let value = item.value else { continue }
+                
+                switch ConfigParam(rawValue: item.name) {
+                case .Secure: secure = value.cldAsBool()
+                case .CdnSubdomain: cdnSubdomain = value.cldAsBool()
+                case .SecureCdnSubdomain: secureCdnSubdomain = value.cldAsBool()
+                case .LongUrlSignature: longUrlSignature = value.cldAsBool()
+                case .SignatureAlgorithm: signatureAlgorithm = SignatureAlgorithm(stringValue: value)
+                case .CName: cname = value
+                case .UploadPrefix: uploadPrefix = value
+                case .Timeout: timeout = value.cldAsNSNumber()
+                default:
                     continue
                 }
-                else {
-                    if let key = ConfigParam(rawValue: keyValue[0]) {
-                        switch (key) {
-                        case .Secure: secure = keyValue[1].cldAsBool()
-                        case .CdnSubdomain: cdnSubdomain = keyValue[1].cldAsBool()
-                        case .SecureCdnSubdomain: secureCdnSubdomain = keyValue[1].cldAsBool()
-                        case .LongUrlSignature: longUrlSignature = keyValue[1].cldAsBool()
-                        case .CName: cname = keyValue[1]
-                        case .UploadPrefix: uploadPrefix = keyValue[1]
-                        case .Timeout: timeout = keyValue[1].cldAsNSNumber()
-                            
-                        default:
-                            break
-                        }
-                    }
-                }
             }
         }
     }
     
     // MARK: Optional Url Params
-    
     internal enum ConfigParam: String, CustomStringConvertible {
-        case Secure =               "secure"
-        case CdnSubdomain =         "cdn_subdomain"
-        case SecureCdnSubdomain =   "secure_cdn_subdomain"
-        case LongUrlSignature =     "long_url_signature"
-        case CName =                "cname"
-        case UploadPrefix =         "upload_prefix"
         
-        case APIKey =               "api_key"
-        case APISecret =            "api_secret"
-        case CloudName =            "cloud_name"
-        case PrivateCdn =           "private_cdn"
-        case SecureDistribution =   "secure_distribution"
-        case Timeout =              "timeout"
+        case Secure             = "secure"
```

Comment:
Again same issue - I need to manually go one by one to see what was changed, and this also looks bad in history - whenever someone looks at config they will see the change was done in a single unrelated commit (this one).